src/serialized-analysis.ts

```diff
@@ -0,0 +1,425 @@
+
+  /** The events that this element fires. */
+  events?: Event[];
+
+  /** The shadow dom content slots that this element accepts. */
+  'slots': Slot[];
+
+  /** Information useful for styling the element and its children. */
+  styling: {
+
+    /** CSS Classes that produce built-in custom styling for the element. */
+    classes: {
+      /** The name of the class. e.g. `bright`, `cascade`, `ominous_pulsing` */
+      name: string;
+      /** A markdown description of the class and what effects it has. */
+      description: string;
+    }[];
+
+    /** CSS Variables that the element understands. */
+    cssVariables: {
+
+      /** The name of the variable. e.g. `--header-color`, `--my-element-size`*/
+      name: string;
+
+      /** The type of the variable. Advisory. e.g. `color`, `size` */
+      type?: string;
+
+      /** A markdown description of the variable. */
+      description?: string;
+
+      /**
+       * A markdown description of how the element will fallback if the variable
+       * isn't defined.
+       */
+      fallbackBehavior?: string;
+    }[];
+
+    /** If true, the element must be given an explicit size by its context. */
+    needsExplicitSize?: boolean;
+
+    // Would be nice to document the default styling a bit here, whether it's
+    // display: block or inline or whatever.
+  };
+
+  /**
+   * An extension point for framework-specific metadata, as well as any
+   * metadata not yet standardized here such as what polyfills are needed,
+   * behaviors and mixins used, the framework that the element was written in,
+   * tags/categories, links to specs that the element implements, etc.
+   */
+  metadata: Object;
+}
+
+export interface Attribute {
+  /** The name of the attribute. e.g. `value`, `icon`, `should-collapse`. */
+  name: string;
+
+  /** A markdown description for the attribute. */
+  description?: string;
+
+  /**
+   * The type that the attribute will be serialized/deserialized as.
+   *
+   * e.g. `string`, `number`, `boolean`, `RegExp`, `Array`, `Object`.
+   */
+  type?: string;
+
+  /** The default value of the attribute, if any. */
+  defaultValue?: string;
+}
+
+export interface Property {
+  /** The name of the property. e.g. `value`, `icon`, `shouldCollapse`. */
+  name: string;
+
+  /** A markdown description of the property. */
+  description: string;
+
+  /**
+   * The javascript type of the property.
+   *
+   * There's no standard here. Common choices are closure compiler syntax
+   * and typescript syntax.
+   */
+  type: string;
+
+  /** A string representation of the default value. */
+  defaultValue?: string;
+
+  /** Nested subproperties hanging off of this property. */
+  properties?: Property[];
+
+  /**
+   * An extension point for framework-specific metadata, as well as any
+   * metadata not yet standardized here such as associated events and
+   * observers.
+   */
+  metadata?: Object;
+}
+
+export interface Event {
+  /** The name of the event. */
```

Comment:
We also need some way to associate a change event with a property or attribute...